components/FileUploader.js

```diff
@@ -114,39 +114,45 @@ class Filename extends Component {
   };
 
   render() {
-    const { iconDescription, status, style, tabIndex, onKeyDown, ...other } = this.props;
-    const tempStyle = Object.assign(style, { marginRight: '-1px' }); // temp style correction for loading component position
-    return (
-      <span>
-        {status === 'uploading'
-          ? <div className="bx--loading" style={tempStyle} {...other}>
-              <svg className="bx--loading__svg" viewBox="-42 -42 84 84">
-                <circle cx="0" cy="0" r="37.5" />
-              </svg>
-            </div>
-          : null}
-        {status === 'edit'
-          ? <Icon
-              className="bx--file-close"
-              name="close--glyph"
-              description={iconDescription}
-              style={style}
-              tabIndex={tabIndex}
-              onKeyDown={onKeyDown}
-              {...other}
-            />
-          : null}
-        {status === 'complete'
-          ? <Icon
-              className="bx--file-complete"
-              name="checkmark--glyph"
-              description={iconDescription}
-              style={style}
-              {...other}
-            />
-          : null}
-      </span>
-    );
+    const { iconDescription, status, style, ...other } = this.props;
+
+    if (status === 'uploading') {
+      return (
+        <div
+          className="bx--loading"
+          style={Object.assign(style, { width: '1rem', height: '1rem' })}
+          {...other}
+        >
+          <svg className="bx--loading__svg" viewBox="-42 -42 84 84">
+            <circle cx="0" cy="0" r="37.5" />
+          </svg>
+        </div>
+      );
+    } else if (status === 'edit') {
+      return (
+        <Icon
+          iconDescription={iconDescription}
```

Comment:
Getting the following trying out the change:

```
Warning: Unknown prop `iconDescription` on <svg> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
    in svg (created by Icon)
    in Icon (created by Filename)
    in Filename (created by FileUploader)
    in span (created by FileUploader)
    in span (created by FileUploader)
    in div (created by FileUploader)
    in div (created by FileUploader)
    in FileUploader (created by Container)
    in div (created by Story)
    in div (created by Story)
    in Story (created by Container)
    in div (created by Container)
    in Container
```

Is there a upcoming corresponding change in `<Icon>`…? (Quickly checked `master` branch as well as PRs but couldn’t find one)

Reply:
OK thanks @hellobrian for checking!